src/utils/misc.ts

```diff
@@ -79,6 +80,35 @@ export function logWarning(msg: string, error?: Error): void {
   }
 }
 
+/**
+   * Logs an error, attempts to unlink the destination file, and exits.
+   *
+   * @param prefix - The message prefix
+   * @param msg - The error message
+   * @param err - The original error
+   * @param destFilePath - The output file path
+   */
+export async function writeError(prefix: string, msg: string, err: Error, destFilePath?: string): Promise<void> {
+  let processedPrefix = prefix;
+  if (!prefix.endsWith(' ')) {
+    processedPrefix += ' ';
+  }
+
+  logError(processedPrefix + msg, err);
+  try {
+    if (destFilePath) {
+      await filesystem.unlink(destFilePath);
+    }
+  } catch (unlinkError) {
+    logError(`${processedPrefix}Failed to unlink mangled file.`, unlinkError);
+  }
+
+  // unless the watcher is active, exit
+  if (!global.snaps.isWatching) {
+    process.exit(1);
+  }
+}
+
```

Comment:
What motivated this move?

Reply:
I felt it was fitting to be in the utility section, and I put it here so I could test it. Unit testing is hard when functions in the same module use each other, as I can't mock/spyOn them.